benchmarks/nx-cugraph/pytest-based/bench_algos.py

```diff
@@ -123,90 +149,308 @@ def get_legacy_backend_selector(backend_name):
+    G = get_graph_obj_for_benchmark(graph_obj, backend_wrapper)
+    result = benchmark.pedantic(
+        target=backend_wrapper(nx.katz_centrality),
+        args=(G,),
+        kwargs=dict(
+            normalized=normalized,
+        ),
+        rounds=rounds,
+        iterations=iterations,
+        warmup_rounds=warmup_rounds,
+    )
+    assert type(result) is dict
+
+
+def bench_k_truss(benchmark, graph_obj, backend_wrapper):
+    G = get_graph_obj_for_benchmark(graph_obj, backend_wrapper)
+    # DiGraphs are not supported
+    if G.is_directed():
+        G = G.to_undirected()
+    result = benchmark.pedantic(
+        target=backend_wrapper(nx.k_truss),
+        args=(G,),
+        kwargs=dict(
+            k=2,
+        ),
+        rounds=rounds,
+        iterations=iterations,
+        warmup_rounds=warmup_rounds,
+    )
+    # Check that this at least appears to be some kind of NX-like Graph
+    assert hasattr(result, "has_node")
+
+
+def bench_out_degree_centrality(benchmark, graph_obj, backend_wrapper):
+    G = get_graph_obj_for_benchmark(graph_obj, backend_wrapper)
+    result = benchmark.pedantic(
+        target=backend_wrapper(nx.out_degree_centrality),
+        args=(G,),
+        rounds=rounds,
+        iterations=iterations,
+        warmup_rounds=warmup_rounds,
+    )
+    assert type(result) is dict
+
+
+def bench_pagerank(benchmark, graph_obj, backend_wrapper):
+    G = get_graph_obj_for_benchmark(graph_obj, backend_wrapper)
+    result = benchmark.pedantic(
+        target=backend_wrapper(nx.pagerank),
+        args=(G,),
+        rounds=rounds,
+        iterations=iterations,
+        warmup_rounds=warmup_rounds,
+    )
+    assert type(result) is dict
+
+
+def bench_single_source_shortest_path_length(benchmark, graph_obj, backend_wrapper):
+    # Use the node with the highest degree
+    degrees = graph_obj.degree()  # list of tuples of (node, degree)
+    node = max(degrees, key=lambda t: t[1])[0]
+    G = get_graph_obj_for_benchmark(graph_obj, backend_wrapper)
+
+    result = benchmark.pedantic(
+        target=backend_wrapper(nx.single_source_shortest_path_length),
+        args=(G,),
+        kwargs=dict(
+            source=node,
+        ),
+        rounds=rounds,
+        iterations=iterations,
+        warmup_rounds=warmup_rounds,
+    )
+    assert type(result) is dict
+
+
+def bench_single_target_shortest_path_length(benchmark, graph_obj, backend_wrapper):
+    # Use the node with the highest degree
+    degrees = graph_obj.degree()  # list of tuples of (node, degree)
+    node = max(degrees, key=lambda t: t[1])[0]
+    G = get_graph_obj_for_benchmark(graph_obj, backend_wrapper)
+
+    # Ensure the benchmark time includes computation of each result from the
+    # returned generator
+    def run_generator(*args, **kwargs):
+        func = backend_wrapper(nx.single_target_shortest_path_length)
+        results = func(*args, **kwargs)
+        return list(results)
```

Comment:
There will be more functions that return generators. Is there a good way to generalize this pattern? Perhaps add e.g. `is_generator=True` to `backend_wrapper`?

Reply:
Good suggestion, done in https://github.com/rapidsai/cugraph/pull/4065/commits/7ac6e314ad58d5980b32195134d252cd4d51ba6b